packages/react-reconciler/src/ReactFiberBeginWork.new.js

```diff
@@ -3376,6 +3341,21 @@ export function checkIfWorkInProgressReceivedUpdate() {
   return didReceiveUpdate;
 }
 
+function resetIndeterminatePlacementForLegacyMode(current, workInProgress) {
+  if ((workInProgress.mode & ConcurrentMode) === NoMode) {
+    if (current !== null) {
```

Comment:
Since this is only relevant to mounts I would hoist our this `current !== null` check. And name it something more like `resetSuspendedCurrentOnMountInLegacyMode`.

Reply:
Done 👍